### PR TITLE
feat: refactor admin component

### PR DESCRIPTION
**Pull request by stefanpoensgen**
https://github.com/tinect/TinectRedirects/issues/17

### CODE REVIEW

**Comment by tinect** [comment]
Thank you, @stefanpoensgen 

When filtering by term, the request is sent twice. Do you have an idea?

File: src/Resources/app/administration/src/module/tinect-redirects/page/tinect-redirects-index/template.html.twig [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                                show-clearable-button
+                        ></sw-entity-single-select>
+
+                        <sw-switch-field :label="$t('tinect-redirects.sidebar.filter.hidden.label')"
+                                         v-model="filter.hidden"/>
+
+                        <sw-button
+                                variant="ghost"
+                                @click="resetFilter">
+                            {{ $tc('tinect-redirects.sidebar.filters.resetFilter') }}
```
Comment (tinect) [inline src/Resources/app/administration/src/module/tinect-redirects/page/tinect-redirects-index/template.html.twig:113]:
```suggestion
                            {{ $tc('tinect-redirects.sidebar.filter.resetFilter') }}
```

**Review by tinect — CHANGES_REQUESTED** [review]

**Comment by stefanpoensgen** [comment]
> Additionally, How does this solve #14 ? I don't see any different.
nvm. it does not ^^

**Comment by stefanpoensgen** [comment]
> When filtering by term, the request is sent twice. Also any detail page is sending it twice. Do you have an idea?

@tinect fixed

File: src/Resources/app/administration/src/module/tinect-redirects/page/tinect-redirects-index/index.js [hunk 2]
```diff
@@ -112,12 +95,28 @@ Component.register('tinect-redirects-index', {
                 criteria.addFilter(Criteria.equals('active', this.filter.active));
             }
 
-            criteria.addFilter(Criteria.equals('hidden', this.filter.hidden));
+            if (this.filter.hidden) {
+                criteria.addFilter(Criteria.equals('active', this.filter.hidden));
+            }
```
Comment (tinect) [inline src/Resources/app/administration/src/module/tinect-redirects/page/tinect-redirects-index/index.js:100]:
```suggestion
            criteria.addFilter(Criteria.equals('hidden', this.filter.hidden));
```

This change needs to be reverted. You filter on active, and this filter should always trigger, while default should not show hidden contents

**Review by tinect — CHANGES_REQUESTED** [review]

**Pull request merged**